comRamona/assignment2/Question1.java

```diff
@@ -0,0 +1,91 @@
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Given a binary tree and a key, write a function that prints all the ancestors of the key in the given
+ * binary tree.
+ */
+public class Question1 {
+    /**
+     * Helper method to find the path to a node and append nodes on the path to the resultPath list(mutable)
+     *
+     * @param node       current root of search
+     * @param target     target value
+     * @param resultPath list to append path
+     * @param <T>        data type of the nodes
+     * @return
+     */
+
+    private static <T> MyNode findAncestorsHelper(MyNode<T> node, MyNode<T> target, List<T> resultPath) {
+        if (node == null) {
+            return null;
+        }
+        if (node.getData().equals(target.getData())) {
+            return node;
+        }
+        MyNode leftSearch = findAncestorsHelper(node.getLeft(), target, resultPath);
+        if (leftSearch != null) {
+            resultPath.add(node.getData());
+            return node;
+        }
+        MyNode rightSearch = findAncestorsHelper(node.getRight(), target, resultPath);
+        if (rightSearch != null) {
+            resultPath.add(node.getData());
+            return node;
+        }
+        return null;
+    }
+
+    /**
+     * @param root   Root node of the tree
+     * @param target Target value to find
+     * @param <T>    data type of the nodes
+     * @return list of ancestors of the target node, null if node not found
+     */
+    public static <T> List<T> findAncestors(MyNode<T> root, MyNode<T> target) {
```

Comment:
It is more likely the target would be a value of type T instead of a Node. As this expects the caller to know all the nodes within the tree before calling.